app/tests/semaphore_test.cc

```diff
@@ -76,6 +84,50 @@ TEST(SemaphoreTest, TimedWait) {
       0.20 * firebase::internal::kMillisecondsPerSecond);
 }
 
+#if FIREBASE_PLATFORM_ANDROID || FIREBASE_PLATFORM_LINUX
+// Use a global variable for `sigusr1_received` because there is no way to pass
+// a context to the signal handler (https://stackoverflow.com/q/64713130).
+std::atomic<bool> sigusr1_received;
+// Tests that Timed Wait handles spurious wakeup (Linux/Android specific).
+TEST(SemaphoreTest, TimedWaitSpuriousWakeupLinux) {
+  // Register a dummy signal handler for SIGUSR1; otherwise, sending SIGUSR1
+  // later on will crash the application.
+  sigusr1_received.store(false);
+  signal(SIGUSR1, [](int signum) { sigusr1_received.store(true); });
+
+  // Start a thread that will send SIGUSR1 to this thread in a few moments.
+  pthread_t main_thread = pthread_self();
+  firebase::Thread signal_sending_thread = firebase::Thread(
+      [](void* arg) {
+        firebase::internal::Sleep(500);
+        pthread_kill(*static_cast<pthread_t*>(arg), SIGUSR1);
+      },
+      &main_thread);
+
+  // Call Semaphore::TimedWait() and keep track of how long it blocks for.
+  firebase::Semaphore sem(0);
+  auto start_ms = firebase::internal::GetTimestamp();
+  const int timed_wait_timeout = 2 * firebase::internal::kMillisecondsPerSecond;
```

Comment:
I think the constants should probably follow the kConstantNaming syntax

Reply:
Done.